domains/pallets/messenger/src/lib.rs

```diff
@@ -935,7 +940,24 @@ mod pallet {
                 if let Some(owner) = &channel.maybe_owner {
                     let hold_id = T::HoldIdentifier::messenger_channel(chain_id, channel_id);
                     let locked_amount = T::Currency::balance_on_hold(&hold_id, owner);
-                    T::Currency::release(&hold_id, owner, locked_amount, Precision::Exact)
+                    let amount_to_release = {
+                        if channel.state == ChannelState::Open {
+                            locked_amount
+                        } else {
+                            let protocol_fee = T::ChannelInitReservePortion::get() * locked_amount;
+                            let release_amount = locked_amount.saturating_sub(protocol_fee);
+                            T::Currency::burn_held(
+                                &hold_id,
+                                owner,
+                                protocol_fee,
+                                Precision::Exact,
+                                Fortitude::Force,
+                            )?;
+                            T::OnXDMRewards::on_xdm_rewards(protocol_fee);
```

Comment:
The protocol fee under the hood will go to the operator of the src chain, if the src chain is a compromised domain that control by a single entity (probably why the dst domain excludes it from the allowlist) then there is no harm to them at all, in this case it makes more sense to also let the operator of the dst chain get this fee since they are the victim.

Reply:
Make sense. Thanks!